### DISCUSSION TITLE
Confusing error message with brew-link for a cask

### DISCUSSION THREAD

**Discussion by jasonkarns** [Homebrew/discussions#3213] [Casks]
Attempting to track down some conflicts between gnupg (formula) and gpg-suite (cask).

Attempted to see what would be linked by `brew link gpg-suite` and so ran with --dry-run flag:

```
$ brew link -n gpg-suite
Found a cask named "gpg-suite" instead. Try
  brew uninstall --cask gpg-suite

Error: No such keg: /usr/local/Cellar/gpg-suite
```

Confusingly, the output of `brew link` suggested I try `brew uninstall`?

Added `--cask` flag, as that seems to be needed for a lot of main brew commands now doubling for casks commands was not recognized as a valid option:

```
$ brew link --cask -n gpg-suite
Usage: brew link, ln [options] installed_formula [...]
```

I'm confused at this point how one can inspect a cask's linkable binaries.

**Comment by gromgit** [comment]
Last I checked, `brew link` works only for formulae, hence the error you got. Any linking done during cask installation (e.g. `binary` stanzas) is permanent and non-optional.

> I'm confused at this point how one can inspect a cask's linkable binaries.

You'd have to look at the cask file itself (`$(brew --repo homebrew/cask)/Casks/gpg-suite.rb` in your example), cross-referencing https://docs.brew.sh/Cask-Cookbook to understand the stanzas therein.

**Reply by SMillerDev** [reply]
`brew cat gpg-suite` would also work

**Comment by carlocab** [comment]
`brew link` works only for formulae. See the [manpage entry](https://docs.brew.sh/Manpage#link-ln-options-installed_formula-) for details.

**Comment by ghost** [comment]
The `google-cloud-sdk` cask may add binaries to itself over time (e.g. `gcloud components install gke-gcloud-auth-plugin`).

Is there a sane way to ensure these are on the homebrew path? For formulas I could `brew unlink && brew link`. 
Or must I manually update them myself? This seems awkward and error prone:

```
ln -sf $HOMEBREW_PREFIX/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin/* $HOMEBREW_PREFIX/bin
```

**Reply by SMillerDev** [reply]
You can just add that folder to the PATH
